src/evaluators/Program.js

```diff
@@ -269,7 +269,7 @@ export default function(ast: BabelNodeProgram, strictCode: boolean, env: Lexical
       let e = realm.getCapturedEffects();
       realm.stopEffectCaptureAndUndoEffects(res);
       // The global state is now at the point where the last fork occurred.
-      if (res.containsCompletion(ThrowCompletion)) {
+      if (res.containsCompletion(ThrowCompletion, true)) {
```

Comment:
I'm a little suspicious that this is the only place that needs to be changed, so I would mildly prefer having a new property on ThrowCompletions that gets checked in containsCompletion so that if we run into another case where we need to check it, it's obvious what went wrong when looking in the debugger.

Reply:
Add another property to the existing one I added? I'm a little confused, please can you expand more?

Reply:
Ah, I misread something. I guess what I'm asking for then is an invariant in serialization or visiting somewhere that ensures that we only ever serialize these `ThrowCompletions` via the `emitThrow` generator entry and not by recursively visiting some `PossiblyNormalCompletion` or `ForkedAbruptCompletion`. Alternatively, you might be able to put the invariant in `join` to make sure we never join a `createdInPureScope` completion.